main.py

```diff
@@ -1,91 +1,131 @@
+    df_tblNotes = df_tblNotes[df_tblNotes['Competition'] == '2e Coupe - CPS']
 
-    #condition = ((df_tblGymnastes['Categorie'] == "Niveau 2A") | (df_tblGymnastes['Categorie'] == "National Ouvert") )
-    categorieSelected = getcategory(df_tblGymnastes)
-    df_tblGymnastes = df_tblGymnastes[categorieSelected]
+    # select lines for current category
+    categorie_selected = get_category(df_tblGymnastes)
+    df_tblGymnastes = df_tblGymnastes[categorie_selected]
 
     # inner join the 2 tables on idGymnaste
     df_joinedTables = df_tblNotes.join(df_tblGymnastes.set_index('idGymnaste'),
-    how = 'inner',on='idGymnaste')
+                                       how='inner', on='idGymnaste')
 
     # add the column total
     df_joinedTables['Total'] = (df_joinedTables['sol'] + df_joinedTables['arcon'] +
-    df_joinedTables['anneaux'] + df_joinedTables['Saut'] +
-    df_joinedTables['parallele'] + df_joinedTables['fixe'])
+                                df_joinedTables['anneaux'] + df_joinedTables['Saut'] +
+                                df_joinedTables['parallele'] + df_joinedTables['fixe'])
+    # sort dataframe by Name and firstname
     df_joinedTables.sort_values(by=['Nom', 'Prenom'], inplace=True)
+    # concatenate name and firstname
     df_joinedTables['Nom'] = df_joinedTables['Prenom'] + ' ' + df_joinedTables['Nom'].str.upper()
 
-    ### 3/ send csv file to google spreadsheet via API to be displayed
-    scope = ['https://spreadsheets.google.com/feeds',
-        'https://www.googleapis.com/auth/drive']
+    return df_joinedTables
+
+
+def send_to_google_spreadsheet_via_api(df_joinedTables):
+    # 3/ send csv file to google spreadsheet via API to be displayed
+    scope = ['https://spreadsheets.google.com/feeds', 'https://www.googleapis.com/auth/drive']
     credentials = ServiceAccountCredentials.from_json_keyfile_name('MyProject8533.json', scope)
     gc = gspread.authorize(credentials)
-    #open an existing spreadsheet
+    # open an existing spreadsheet
     sheet = gc.open('ResultatsCoupeQuebec')
     ws = sheet.get_worksheet(0)
-    #clear all values before updating
+    # clear all values before updating
     ws.clear()
-    #find the range of cells with the size of the dataframe
+    # find the range of cells with the size of the dataframe
     cell_list = ws.range('A1:I' + str(df_joinedTables.shape[0]))
 
-    cell_values = df_joinedTables[['Nom', 'NomClub','sol','arcon', 'anneaux',
-    'Saut', 'parallele', 'fixe', 'Total']].values.flatten()
+    cell_values = df_joinedTables[['Nom', 'NomClub', 'sol', 'arcon', 'anneaux',
+                                   'Saut', 'parallele', 'fixe', 'Total']].values.flatten()
 
-    for counter, val in enumerate(cell_values):  #gives us a tuple of an index and value
-        cell_list[counter].value = val    #use the index on cell_list and the val from cell_values
+    for counter, val in enumerate(cell_values):  # gives us a tuple of an index and value
```

Comment:
alors: oui aux commentaires, mais aux commentaires utiles 😊 
` # gives us a tuple of an index and value` c'est litteralement la documentation de enumerate. donc c'est pas vraiment necessaire

Reply:
ha ha, en effet, c'était surtout pour moi, pour m'en rappeler parce que je ne connaissais pas cette fonction. mais oui tu as raison ;-)